pilot/pkg/model/jwks_resolver.go

```diff
@@ -126,7 +179,93 @@ func (r *jwksURIResolver) resolveJwksURIUsingOpenID(issuer string) (string, erro
 	}
 
 	// Set JwksUri in cache.
-	r.cache.Set(issuer, jwksURI)
+	r.JwksURICache.Set(issuer, jwksURI)
 
 	return jwksURI, nil
 }
+
+func (r *jwksResolver) fetchJwtPubKey(jwksURI string) (string, error) {
```

Comment:
Probably, have a function for fetching remote content, and reuse in both resolveJwksURIUsingOpenID and here. In fact, this function is basically that.

Reply:
Since parsing response part is different for uri/key, the shared code isn't that much; I may prefer to keep as it is for readability(ex, defer response.body.close)

Reply:
It's not different until you parse the content. I would name this function as `getRemoteContent(url string)`, and call it in your resolveJwksURIUsingOpenID, eg:
```go
func resolveJwksURIUsingOpenID(issuer string) (string, error) {
  body, err := getRemoteContent(issuer)
  ...
  // parse body and return
}
```

It also avoid having very similar function name (GetPublicKey and fetchPublicKey)